decidim-admin/app/queries/decidim/admin/manageable_participatory_processes_for_user.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+module Decidim
+  module Admin
+    # A class used to find the ParticipatoryProcesses that the given user can
+    # manage.
+    class ManageableParticipatoryProcessesForUser < Rectify::Query
+      # Initializes the class.
+      #
+      # user - a User that needs to find which processes can manage
+      def initialize(user)
+        @user = user
+      end
+
+      # Finds the ParticipatoryProcesses that the given user can manage.
+      #
+      # Returns an ActiveRecord::Relation.
+      def query
+        return user.organization.participatory_processes if user.role?(:admin)
+
+        user.participatory_processes
```

Comment:
this can be extra-refactored to use a single query, @oriolgual is the ActiveRecord master